pkg/controller/container/cluster.go

```diff
@@ -179,24 +184,27 @@ func (e *clusterExternal) Update(ctx context.Context, mg resource.Managed) (reso
 	if !ok {
 		return resource.ExternalUpdate{}, errors.New(errNotCluster)
 	}
-
+	// Do not issue another update until the cluster finishes the previous one.
+	if cr.Status.AtProvider.Status == v1beta1.ClusterStateReconciling || cr.Status.AtProvider.Status == v1beta1.ClusterStateProvisioning {
```

Comment:
@muvaf I added a condition here not to `Update` if in `PROVISIONING` state either because we start seeing errors in the conditions due to the fact that we create with the bootstrap node pool and then immediately try to delete it before the cluster is even ready. I think this is good in the general case as well to not try and update when a resource is in the creation phase.

Reply:
One thing to note is that sometimes we still get errors on the `NodePool` for operations because the updates are coupled with the cluster, so if we issue another request before the status of the node pool changes then we could be given back an error such as: `delete failed: cannot delete GKE node pool: googleapi: Error 400: Operation operation-1576863807942-a13daf3f is currently deleting a node pool for cluster gkev1beta1-cool. Please wait and try again once it is done., failedPrecondition`. The node pool itself is actually being deleted but its status does not show it yet.

Reply:
> if we issue another request before the status of the node pool changes then we could be given back an error

Got it. Though I don't see there is much we can do in this case, right?

Reply:
That is my thought as well.